client/src/main/java/com/tigrisdata/db/client/TigrisAsyncDatabase.java

```diff
@@ -50,27 +60,38 @@ CompletableFuture<CreateOrUpdateCollectionsResponse> createOrUpdateCollections(
    * @return future to the {@link CreateOrUpdateCollectionsResponse}
    */
   CompletableFuture<CreateOrUpdateCollectionsResponse> createOrUpdateCollections(
-      String[] packagesToScan, Optional<Predicate<Class<? extends TigrisCollectionType>>> filter);
+      String[] packagesToScan,
+      Optional<Predicate<Class<? extends TigrisDocumentCollectionType>>> filter);
 
   /**
    * Drops the collection.
    *
-   * @param collectionTypeClass Class type of the collection
-   * @param <T> type of the collection that is of type {@link TigrisCollectionType}
+   * @param documentCollectionTypeClass Class type of the collection
+   * @param <T> type of the collection that is of type {@link TigrisDocumentCollectionType}
    * @return the future to the {@link DropCollectionResponse}
    */
-  <T extends TigrisCollectionType> CompletableFuture<DropCollectionResponse> dropCollection(
-      Class<T> collectionTypeClass);
+  <T extends TigrisDocumentCollectionType> CompletableFuture<DropCollectionResponse> dropCollection(
```

Comment:
Do we have drop for Topic as well?

Reply:
discussed offline - we will add drop topic later on.